### DISCUSSION TITLE
how to apply dynamic textStyle?

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/panda#2416] [Q&A]
I want to apply the textstyle differently depending on the props
- first
![스크린샷 2024-03-22 오후 2 56 41](https://github.com/chakra-ui/panda/assets/147475576/4bb4ad9b-d15f-4842-b25b-c71afa114e3e)
It doesn't seem to be applicable to the overall textstyle, it seems to be only a part of it

- second
![스크린샷 2024-03-22 오후 2 53 30](https://github.com/chakra-ui/panda/assets/147475576/b9c35009-1adf-4900-944f-eec0c4d7213e)
![스크린샷 2024-03-22 오후 2 53 37](https://github.com/chakra-ui/panda/assets/147475576/f5290801-71c7-4bf8-a5f1-4e30bc2b8fea)

I want to declare var, but I get an error message.

Is there a way?

**Comment by ghost** [comment]
export const dynamicFontStyle = cva({
  variants: {
    fontStyle: {
      title100: {
        fontSize: '24px',
        lineHeight: '32px',
      },
      title200: {
        fontSize: '20px',
        lineHeight: '28px',
      },
      body100: {
        fontSize: '16px',
        lineHeight: '24px',
      },
      body200: {
        fontSize: '14px',
        lineHeight: '20px',
      },
      caption100: {
        fontSize: '12px',
        lineHeight: '16px',
      },
      caption200: {
        fontSize: '10px',
        lineHeight: '16px',
      },
      label100: {
        fontSize: '14px',
        lineHeight: '20px',
      },
      label200: {
        fontSize: '12px',
        lineHeight: '16px',
      },
    },
  },
});

Should I create and use cva in this way?

**Reply by astahmer** [reply]
yes that can work

**Comment by astahmer** [comment]
in your first screenshot, you renamed the `textStyle` property to `fontStyle`, this is most likely why it wasn't working as expected as [this is a known limitation](https://panda-css.com/docs/concepts/style-props#property-renaming) with the static analysis

**Reply by ghost** [reply]
@astahmer 
why occure error style `--badge` ?

**Reply by ghost** [reply]
What do I do to prevent an error from occurring?

**Reply by astahmer** [reply]
sorry I don't understand the question, can you show an example ?

**Reply by ghost** [reply]
Sorry I wasn't specific. In the second picture in the text, I'm trying to assign a value to the style under the name of '--badge', but an error like that occurs

**Reply by astahmer** [reply]
ah yes, this is just react and its typings, it's not related to Panda and it's a type only error (from Typescript), it works fine at runtime (in your browser)

**Reply by ghost** [reply]
I've confirmed that it's working. However, I can't build it due to an error at the time of build, so is there a solution?

**Reply by ghost** [reply]
i solved this error
```ts
style={{
'--badge':'test'
} as React.CSSProperties}
```
Is this the best way?
